python/vyos/ifconfig/erspan.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019-2020 VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+# https://developers.redhat.com/blog/2019/05/17/an-introduction-to-linux-virtual-interfaces-tunnels/#erspan
+# http://vger.kernel.org/lpc_net2018_talks/erspan-linux-presentation.pdf
+
+from copy import deepcopy
+
+from netaddr import EUI
+from netaddr import mac_unix_expanded
+from random import getrandbits
+
+from vyos.util import dict_search
+from vyos.ifconfig.interface import Interface
+from vyos.validate import assert_list
+
+@Interface.register
+class _ERSpan(Interface):
+    """
+    _ERSpan: private base class for ERSPAN tunnels
+    """
+    default = {
+        **Interface.default,
+        **{
+            'type': 'erspan',
+        }
+    }
+    definition = {
+        **Interface.definition,
+        **{
+            'section': 'erspan',
+            'prefixes': ['ersp',],
+        },
+    }
+    
+    options = ['local_ip','remote_ip','encapsulation','parameters']
+    
+    def __init__(self,ifname,**config):
+        self.config = deepcopy(config) if config else {}
+        super().__init__(ifname, **self.config)
+    
+    def change_options(self):
```

Comment:
I wonder why there can't be a common change function as there is the `type` field in `self.config` which should either hold `erspan` or `ip6erspan`, please consult the existing implementations of gre or ipip.

The code looks too complex to me at first glance.

Reply:
Hey, when I detect that an interface has changed, I need to delete the previous interface first

Reply:
In the existing implementation of tunnel, when the package type is changed, there is a same implementation, I just copied this implementation
https://github.com/vyos/vyos-1x/blob/0b06b4807887357b2b9eed6a35928bb343c41028/src/conf_mode/interfaces-tunnel.py#L129-L135

https://github.com/vyos/vyos-1x/blob/0b06b4807887357b2b9eed6a35928bb343c41028/src/conf_mode/interfaces-tunnel.py#L65-L66